### PR TITLE
add broker load internal doc

### PR DESCRIPTION
**Pull request by kangpinghuang**
add a doc of broker load realization.

### CODE REVIEW

File: docs/documentation/cn/internal/broker_load.md [hunk 1]
```diff
@@ -0,0 +1,1014 @@
...
+        _schema->column(i)->consume(
+                &cell, (const char*)value, is_null, mem_pool, &_agg_object_pool);
+    }
+}
+```
+DeltaWriter::write()会通过调用MemTable::insert（）将数据写入到MemTable中，上述函数完成将base表的数据转为rollup表的数据的逻辑。其中，insert参数tuple是base表的数据，_tuple_to_row具体完成了将tuple转化成rollup index的数据。并且底层使用跳表保存数据集合，会自动对数据行进行排序。如果是聚合模型，还会对数据进行聚合计算。
+
+4. 如何实现导入的原子性
+
+	在LoadLoadingTask任务下发之后，就会开始事务在BE的中执行，并且在BE中DeltaWriter会调用prepare_txn将事务进行持久化。如果所有的LoadLoadingTask quorum完成，则会认为该事务成功，会通过publish_version rpc向BE发送PublishVersionTask，进行事务的提交。
```
Comment (EmmyMiao87) [inline docs/documentation/cn/internal/broker_load.md]: task quorum完成稍微可能会被误解，应该是导入涉及到的所有tablet都quorum完成

**Review by EmmyMiao87 — CHANGES_REQUESTED** [review]

File: docs/documentation/cn/internal/broker_load.md [hunk 2]
```diff
@@ -0,0 +1,1014 @@
+# Doris Broker导入实现解析
```
Comment (imay) [inline docs/documentation/cn/internal/broker_load.md]: license header

**Review by imay — CHANGES_REQUESTED** [review]

File: docs/documentation/cn/internal/broker_load.md (hunk 2 above)
Comment (EmmyMiao87) [reply docs/documentation/cn/internal/broker_load.md]: Long time no see, Doctor Zhao~

**Comment by kangkaisen** [comment]
@kangpinghuang  Great doc. Thanks for your share.

**Comment by kangkaisen** [comment]
One suggestion:
**Please remove the most of code, give the method name is enough，we should use language and picture to instead of the code.**

**Review by morningman — CHANGES_REQUESTED** [review]
建议整个 broker 导入主要重点介绍 FE 端的作业调度和任务生成部分。
BE端的 BrokerScanNode -> OlapTableSink 部分单独作为整个流式导入框架的BE端部分单独行程一篇文档。

File: docs/documentation/cn/internal/broker_load.md [hunk 3]
```diff
@@ -0,0 +1,1014 @@
...
+
+## 名词解释
+
+* FE：Frontend，即 Palo 的前端节点。主要负责接收和返回客户端请求、元数据以及集群管理、查询计划生成等工作。关于Doris的架构图，参考[Doris架构介绍](http://doris.incubator.apache.org/)
+* BE：Backend，即 Palo 的后端节点。主要负责数据存储与管理、查询计划执行等工作。
+* Broker：请参考[Broker文档](http://doris.incubator.apache.org/documentation/cn/administrator-guide/broker.html)
+
+## 实现原理
+
+在Broker导入中，用户只要提供一份base表的数据，Doris会为用户进行一下处理：
```
[the remark this answers is not in the document]
Comment (kangpinghuang) [reply docs/documentation/cn/internal/broker_load.md]: ok

File: docs/documentation/cn/internal/broker_load.md [hunk 4]
```diff
@@ -0,0 +1,1014 @@
...
++---^---+     +---^---+    +---^---+
+    |             |            | 
++----------------------------------+
+|       HDFS/BOS/AFS cluster       |
++----------------------------------+
+```
+
+整个导入过程大体如下：
+
+- 用户将请求发送到FE，经过FE进行语法和语意分析，之后生成BrokerLoadJob
```
[the remark this answers is not in the document]
Comment (kangpinghuang) [reply docs/documentation/cn/internal/broker_load.md]: ok

File: docs/documentation/cn/internal/broker_load.md [hunk 5]
```diff
@@ -0,0 +1,1014 @@
...
+        // 通过 DataSink 的 send() 方法发送数据
+        RETURN_IF_ERROR(_sink->send(runtime_state(), batch));
+    }
+
+    ...
+
+    return Status::OK;
+}
+```
+一个PlanFragment中包含多个ExecNode组成的树状执行计划（类型spark stage中包含的rdd），其中SinkNode(上面中的_sink变量)类型spark中带action操作的rdd，执行数据序列化操作，可能后续会进行持久化，也可能通过网络发送到下一个PlanFragment中。get_next_internal会调用PlanNode执行计划树获取数据，然后通过SinkNode发送到下一个PlanFragment或者输出结果。
```
[the remark this answers is not in the document]
Comment (kangpinghuang) [reply docs/documentation/cn/internal/broker_load.md]: ok，我修改一下

**Pull request closed without merging**